connectors/src/connectors/google_drive/lib.ts

```diff
@@ -65,34 +76,87 @@ export async function registerWebhook(
   }
 }
 
+export async function isDriveObjectExpandable({
+  objectId,
+  mimeType,
+  connectorId,
+  viewType,
+}: {
+  objectId: string;
+  mimeType: string;
+  connectorId: ModelId;
+  viewType: ContentNodesViewType;
+}): Promise<boolean> {
+  if (isGoogleDriveSpreadSheetFile({ mimeType }) && viewType === "tables") {
+    // In tables view, Spreadsheets can be expanded to show their sheets.
+    return !!(await GoogleDriveSheet.findOne({
+      attributes: ["id"],
+      where: {
+        driveFileId: objectId,
+        connectorId: connectorId,
+      },
+    }));
+  }
+
+  const where: WhereOptions<InferAttributes<GoogleDriveFiles>> = {
+    connectorId: connectorId,
+    parentId: objectId,
+  };
+
+  if (viewType === "tables") {
+    // In tables view, we only show folders and spreadhsheets.
+    // A folder that only contains Documents is not expandable.
+    where.mimeType = [
```

Comment:
`mimeType` is not part of any composite index, this will very likely be slow in production.

Reply:
It is not, but the query is already quite filtered so it won't really be slower than just getting the whole thing without filtering right ? 
I can add the index in a separate PR

Reply:
Let see how it goes in prod, but I already observed some slowness around how we compute the expandable, just want to surface this potential limit!

Reply:
Yes, thanks for doing so !